diff --unified -uNrp cover-thumbnailer_0.8.3_src/cover-thumbnailer.py cover-thumbnailer_0.8.3_src-new/cover-thumbnailer.py
--- cover-thumbnailer_0.8.3_src/cover-thumbnailer.py	2011-10-27 14:26:35.000000000 +0200
+++ cover-thumbnailer_0.8.3_src-new/cover-thumbnailer.py	2013-09-09 11:40:55.210256375 +0200
@@ -49,7 +49,8 @@ __author__ = "Fabien Loison <flo@flogiso
 __copyright__ = "Copyright © 2009 - 2011 Fabien LOISON"
 
 
-import Image, urllib, os.path, sys, re
+from PIL import Image
+import urllib, os.path, sys, re
 
 
 #==================================================================== CONF ====
