diff --git a/extensions/raw_files/gth-metadata-provider-raw.c b/extensions/raw_files/gth-metadata-provider-raw.c
index c05b58c..4dfeec6 100644
--- a/extensions/raw_files/gth-metadata-provider-raw.c
+++ b/extensions/raw_files/gth-metadata-provider-raw.c
@@ -21,7 +21,7 @@
 
 #include <config.h>
 #include <gthumb.h>
-#include <libraw.h>
+#include <libraw/libraw.h>
 #include "gth-metadata-provider-raw.h"
 #include "main.h"
 
diff --git a/extensions/raw_files/main.c b/extensions/raw_files/main.c
index 00a4c77..810f864 100644
--- a/extensions/raw_files/main.c
+++ b/extensions/raw_files/main.c
@@ -44,7 +44,7 @@ const char *raw_mime_types[] = {
 #include <cairo.h>
 #include <gtk/gtk.h>
 #include <gthumb.h>
-#include <libraw.h>
+#include <libraw/libraw.h>
 #include "gth-metadata-provider-raw.h"
 
 
