diff --git a/src/build/GhosttyResources.zig b/src/build/GhosttyResources.zig
index cae907ec..5d59c4c6 100644
--- a/src/build/GhosttyResources.zig
+++ b/src/build/GhosttyResources.zig
@@ -203,7 +203,7 @@ pub fn init(b: *std.Build, cfg: *const Config) !GhosttyResources {
         // Right click menu action for Nautilus
         try steps.append(&b.addInstallFile(
             b.path("dist/linux/ghostty_nautilus.py"),
-            "share/nautilus-python/extensions/com.mitchellh.ghostty.py",
+            "share/nautilus-python/extensions/ghostty.py",
         ).step);
 
         // Various icons that our application can use, including the icon
