From 50234aa43c303ccd05c24a568c684726ae35e2f0 Mon Sep 17 00:00:00 2001
From: Poncho <poncho@spahan.ch>
Date: Wed, 22 Jan 2025 10:06:16 +0100
Subject: [PATCH 2/2] ghostty-1.0.2-copy-terminfo-using-installdir.patch

---
 src/build/GhosttyResources.zig | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/src/build/GhosttyResources.zig b/src/build/GhosttyResources.zig
index cae907ec..51943d7b 100644
--- a/src/build/GhosttyResources.zig
+++ b/src/build/GhosttyResources.zig
@@ -57,14 +57,12 @@ pub fn init(b: *std.Build, cfg: *const Config) !GhosttyResources {
             run_step.step.dependOn(&src_install.step);
 
             {
-                // Use cp -R instead of Step.InstallDir because we need to preserve
-                // symlinks in the terminfo database. Zig's InstallDir step doesn't
-                // handle symlinks correctly yet.
-                const copy_step = RunStep.create(b, "copy terminfo db");
-                copy_step.addArgs(&.{ "cp", "-R" });
-                copy_step.addFileArg(path);
-                copy_step.addArg(b.fmt("{s}/share", .{b.install_path}));
-                try steps.append(&copy_step.step);
+                const install_dir_step = b.addInstallDirectory(.{
+                    .source_dir = path,
+                    .install_dir = .prefix,
+                    .install_subdir = "share/terminfo",
+                });
+                b.getInstallStep().dependOn(&install_dir_step.step);
             }
         }
     }
-- 
2.45.3

