diff --unified --new-file --recursive --show-c-function gplaycli-3.23/setup.py gplaycli-3.23-fix/setup.py
--- gplaycli-3.23/setup.py	2018-05-15 17:17:01.000000000 +0200
+++ gplaycli-3.23-fix/setup.py	2018-07-14 09:05:12.521941851 +0200
@@ -28,7 +28,7 @@ setup(name='GPlayCli',
 			'gplaycli': 'gplaycli',
 		},
 		data_files=[
-			[os.path.expanduser('~')+'/.config/gplaycli', ['gplaycli.conf','cron/cronjob']],
+                        ['/etc/gplaycli/', ['gplaycli.conf','cron/cronjob']],
 		],
 		install_requires=[
 				'gpapi == 0.4.2',
