diff --git a/bash/expert/PolShell b/bash/expert/PolShell
index d2afad0..37777f7 100755
--- a/bash/expert/PolShell
+++ b/bash/expert/PolShell
@@ -20,5 +20,4 @@
 [ "$PLAYONLINUX" = "" ] && exit 0
 source "$PLAYONLINUX/lib/sources"
 
-[ "$DEBIAN_PACKAGE" = "FALSE" ] && TITLEADD="-T $APPLICATION_TITLE"
 "$POL_TERM" $TITLEADD -e "$PLAYONLINUX/bash/polshell"
diff --git a/lib/playonlinux.lib b/lib/playonlinux.lib
index 61144e6..dc60b17 100755
--- a/lib/playonlinux.lib
+++ b/lib/playonlinux.lib
@@ -226,7 +226,7 @@ EOF
 	fi
 	echo bash --rcfile "$POL_USER_ROOT/tmp/env_bash" > "$POL_USER_ROOT/tmp/env_bash2"
 	chmod +x "$POL_USER_ROOT/tmp/env_bash2"
-	xterm -e "$POL_USER_ROOT/tmp/env_bash2" -T "$APPLICATION_TITLE"
+	$POL_TERM -e "$POL_USER_ROOT/tmp/env_bash2" $TITLEADD
 }
 
 
diff --git a/lib/variables b/lib/variables
index 9dbd491..a3728a0 100755
--- a/lib/variables
+++ b/lib/variables
@@ -32,7 +32,8 @@ alias find="find -H"
 if [ "$DEBIAN_PACKAGE" = "TRUE" ]; then
 	export POL_TERM="x-terminal-emulator"
 else
-	export POL_TERM="xterm"
+	export POL_TERM="gnome-terminal"
+	export TITLEADD="-t $APPLICATION_TITLE"
 fi
 
 ### Desktop
diff --git a/python/mainwindow.py b/python/mainwindow.py
index 53921b0..4922a13 100755
--- a/python/mainwindow.py
+++ b/python/mainwindow.py
@@ -1231,7 +1231,7 @@ class PlayOnLinuxApp(wx.App):
         self.singleCheckFatal("gpg", package="GnuPG")
 
         if(os.environ["DEBIAN_PACKAGE"] == "FALSE"):
-            self.singleCheck("xterm")
+            self.singleCheck("gnome-terminal")
         self.singleCheck("gettext.sh", package="gettext")  # gettext-base on Debian
         self.singleCheck("icotool", package="icoutils")
         self.singleCheck("wrestool", package="icoutils")
