diff --unified --new-file --recursive --show-c-function '--color=auto' garminexport-0.4.2-garth/garminexport/backup.py garminexport-0.4.2/garminexport/backup.py
--- garminexport-0.4.2-garth/garminexport/backup.py	2023-09-27 20:47:16.008207589 +0200
+++ garminexport-0.4.2/garminexport/backup.py	2023-09-27 20:48:26.824694214 +0200
@@ -4,6 +4,7 @@ import codecs
 import json
 import logging
 import os
+from datetime import datetime, timezone
 
 log = logging.getLogger(__name__)
 
@@ -42,9 +43,9 @@ def export_filename(activity, export_for
     :return: The file name to use for the exported activity.
     :rtype: str
     """
-    fn = "{time}_{id}{suffix}".format(
-        id=activity[0],
-        time=activity[1].isoformat(),
+    time_local = activity[1].replace(tzinfo=timezone.utc).astimezone(tz=None)
+    fn = "{time}{suffix}".format(
+        time=time_local.strftime('%Y-%m-%d-%H-%M-%S'),
         suffix=format_suffix[export_format])
     return fn.replace(':', '_') if os.name == 'nt' else fn
 
