diff --unified --new-file --recursive --show-c-function '--color=auto' stagit-0.9.3/Makefile stagit-0.9.3-gentoo/Makefile
--- stagit-0.9.3/Makefile	2020-02-23 20:30:46.000000000 +0100
+++ stagit-0.9.3-gentoo/Makefile	2020-03-01 09:54:54.886803162 +0100
@@ -4,12 +4,12 @@ NAME = stagit
 VERSION = 0.9.3
 
 # paths
-PREFIX = /usr/local
-MANPREFIX = ${PREFIX}/man
-DOCPREFIX = ${PREFIX}/share/doc/${NAME}
+PREFIX = /usr
+MANPREFIX = ${PREFIX}/share/man
+DOCPREFIX = ${PREFIX}/share/doc/${P}
 
-LIBGIT_INC = -I/usr/local/include
-LIBGIT_LIB = -L/usr/local/lib -lgit2
+LIBGIT_INC = -I/usr/include
+LIBGIT_LIB = -L/usr/lib -lgit2
 
 # use system flags.
 STAGIT_CFLAGS = ${LIBGIT_INC} ${CFLAGS}
