diff --git a/config.mk b/config.mk
index 90419a0..243b1cb 100644
--- a/config.mk
+++ b/config.mk
@@ -1,15 +1,15 @@
 # customize below to fit your system
 
 # paths
-PREFIX = /usr/local
-MANPREFIX = ${PREFIX}/man
-DOCPREFIX = ${PREFIX}/share/doc/stagit
+PREFIX = /usr
+MANPREFIX = ${PREFIX}/share/man
+DOCPREFIX = ${PREFIX}/share/doc/${P}
 
 # compiler and linker
 #CC = cc
 
-GITINC = /usr/local/include
-GITLIB = /usr/local/lib
+GITINC = /usr/include
+GITLIB = /usr/lib
 
 # includes and libs
 INCS = -I${GITINC}
@@ -20,11 +20,11 @@ LIBS = -L${GITLIB} -lgit2
 #LDFLAGS = ${LIBS}
 
 # optimized
-CFLAGS = -O2 -std=c99 ${INCS}
-LDFLAGS = -s ${LIBS}
+CFLAGS += ${INCS}
+LDFLAGS += ${LIBS}
 
 # optimized static
 #CFLAGS = -static -O2 -std=c99 ${INCS}
 #LDFLAGS = -static -s ${LIBS}
 
-CPPFLAGS = -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_BSD_SOURCE ${INCS}
+CPPFLAGS += -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -D_BSD_SOURCE ${INCS}
