diff --unified --new-file --recursive --show-c-function cnijfilter2-source-5.00-1.orig/cmdtocanonij2/filter/Makefile.am cnijfilter2-source-5.00-1/cmdtocanonij2/filter/Makefile.am
--- cnijfilter2-source-5.00-1.orig/cmdtocanonij2/filter/Makefile.am	2014-07-16 02:55:34.000000000 +0200
+++ cnijfilter2-source-5.00-1/cmdtocanonij2/filter/Makefile.am	2015-11-01 12:18:02.436943646 +0100
@@ -1,4 +1,4 @@
-filterdir=$(libdir)/cups/filter
+filterdir=/usr/libexec/cups/filter
 
 filter_PROGRAMS= cmdtocanonij2
 
diff --unified --new-file --recursive --show-c-function cnijfilter2-source-5.00-1.orig/cnijbe2/src/Makefile.am cnijfilter2-source-5.00-1/cnijbe2/src/Makefile.am
--- cnijfilter2-source-5.00-1.orig/cnijbe2/src/Makefile.am	2014-07-16 02:55:34.000000000 +0200
+++ cnijfilter2-source-5.00-1/cnijbe2/src/Makefile.am	2015-11-01 12:17:17.816145475 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-backend_bindir = /usr/lib/cups/backend
+backend_bindir = /usr/libexec/cups/backend
 backend_bin_PROGRAMS = cnijbe2
 transform = 
 
diff --unified --new-file --recursive --show-c-function cnijfilter2-source-5.00-1.orig/rastertocanonij/src/Makefile.am cnijfilter2-source-5.00-1/rastertocanonij/src/Makefile.am
--- cnijfilter2-source-5.00-1.orig/rastertocanonij/src/Makefile.am	2014-07-16 02:55:34.000000000 +0200
+++ cnijfilter2-source-5.00-1/rastertocanonij/src/Makefile.am	2015-11-01 12:18:24.450670547 +0100
@@ -1,4 +1,4 @@
-filterdir=$(libdir)/cups/filter
+filterdir=/usr/libexec/cups/filter
 
 filter_PROGRAMS= rastertocanonij
 
