diff --git a/gui/map.cc b/gui/map.cc
index 9efeb6e0..4f08477b 100644
--- a/gui/map.cc
+++ b/gui/map.cc
@@ -37,6 +37,7 @@
 #include <QApplication>
 #include <QCursor>
 #include <QFile>
+#include <QStandardPaths>
 #include <QTextStream>
 
 #include <math.h>
@@ -90,8 +91,9 @@ Map::Map(QWidget* parent,
   connect(mclicker, SIGNAL(logTime(QString)), this, SLOT(logTime(QString)));
 #endif
 
-  QString baseFile =  QApplication::applicationDirPath() + "/gmapbase.html";
-  if (!QFile(baseFile).exists()) {
+  QString baseFile =  QStandardPaths::locate(QStandardPaths::GenericDataLocation,
+                                             "gpsbabel/gmapbase.html");
+  if (baseFile.isEmpty()) {
     QMessageBox::critical(nullptr, appName,
                           tr("Missing \"gmapbase.html\" file.  Check installation"));
   } else {
