diff --git a/cmake/admCheckQt5.cmake b/cmake/admCheckQt5.cmake
index 19b56a4..9fd6386 100644
--- a/cmake/admCheckQt5.cmake
+++ b/cmake/admCheckQt5.cmake
@@ -88,10 +88,7 @@ MACRO(checkQt5)
                         MESSAGE(STATUS "  Checking for Qt5OpenGL")
                         FIND_PACKAGE(Qt5OpenGL)    
                         IF(Qt5OpenGL_FOUND)
-                                SET(QT_QTOPENGL_FOUND True)
-                                SET(QT_QTOPENGL_INCLUDE_DIR ${Qt5OpenGL_INCLUDE_DIRS})
-                                SET(QT_QTOPENGL_LIBRARIES   ${Qt5OpenGL_LIBRARIES})
-                                MESSAGE(STATUS "  Found,include=${QT_OPENGL_INCLUDE_DIR}, libs=${QT_OPENGL_LIBRARIES}")
+                                MESSAGE(STATUS "  Manually Disabled")
                         ELSE(Qt5OpenGL_FOUND)
                                 MESSAGE(STATUS "  NOT Found")
                         ENDIF(Qt5OpenGL_FOUND)
