From bf85739a2e70ff7bd5c0b92554d9c6078daedfc1 Mon Sep 17 00:00:00 2001
From: Poncho <poncho@spahan.ch>
Date: Sat, 30 May 2020 13:05:15 +0200
Subject: [PATCH 1/2] Workout: Only scale on Ctrl+mouse wheel

---
 src/Train/WorkoutWidget.cpp | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/src/Train/WorkoutWidget.cpp b/src/Train/WorkoutWidget.cpp
index 73c83251c..ffed2a0a2 100644
--- a/src/Train/WorkoutWidget.cpp
+++ b/src/Train/WorkoutWidget.cpp
@@ -565,15 +565,20 @@ WorkoutWidget::eventFilter(QObject *obj, QEvent *event)
     //
     if (event->type() == QEvent::Wheel) {
 
-        // STATE: NONE
-        if (state == none) {
-            QWheelEvent *w = static_cast<QWheelEvent*>(event);
-            updateNeeded = scale(w->angleDelta());
-            filterNeeded = true;
-        }
+        Qt::KeyboardModifiers kmod = static_cast<QInputEvent*>(event)->modifiers();
+        bool ctrl = (kmod & Qt::ControlModifier) != 0;
+        if (ctrl) {
 
-        // will need to ..
-        recompute();
+            // STATE: NONE
+            if (state == none) {
+                QWheelEvent *w = static_cast<QWheelEvent*>(event);
+                updateNeeded = scale(w->angleDelta());
+                filterNeeded = true;
+            }
+
+            // will need to ..
+            recompute();
+        }
     }
 
     //
-- 
2.26.2

