From 7f0a9aa6ca312ccb5f6ea0ab558bbd27a97817d5 Mon Sep 17 00:00:00 2001
From: Poncho <poncho@spahan.ch>
Date: Tue, 13 Apr 2021 09:50:20 +0200
Subject: [PATCH 3/5] Workout: Only scale on Ctrl+mouse wheel

---
 src/Train/WorkoutWidget.cpp | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/src/Train/WorkoutWidget.cpp b/src/Train/WorkoutWidget.cpp
index a42bcf731..7bbf6d921 100644
--- a/src/Train/WorkoutWidget.cpp
+++ b/src/Train/WorkoutWidget.cpp
@@ -571,15 +571,20 @@ WorkoutWidget::eventFilter(QObject *obj, QEvent *event)
     //
     if (event->type() == QEvent::Wheel) {
 
-        // STATE: NONE
-        if (state == none) {
-            QWheelEvent *w = static_cast<QWheelEvent*>(event);
-            updateNeeded = scale(w->angleDelta());
-            filterNeeded = true;
-        }
+        Qt::KeyboardModifiers kmod = static_cast<QInputEvent*>(event)->modifiers();
+        bool ctrl = (kmod & Qt::ControlModifier) != 0;
+        if (ctrl) {
 
-        // will need to ..
-        recompute();
+            // STATE: NONE
+            if (state == none) {
+                QWheelEvent *w = static_cast<QWheelEvent*>(event);
+                updateNeeded = scale(w->angleDelta());
+                filterNeeded = true;
+            }
+
+            // will need to ..
+            recompute();
+        }
     }
 
     //
-- 
2.26.3

