From d27b04d3cdcca0471bb4d2036d9b3e5fbd04805c Mon Sep 17 00:00:00 2001
From: Poncho <poncho@spahan.ch>
Date: Sat, 9 Nov 2019 10:05:42 +0100
Subject: [PATCH 4/4] src/FileIO/FitRideFile.cpp: add Stages Cycling

---
 src/FileIO/FitRideFile.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/FileIO/FitRideFile.cpp b/src/FileIO/FitRideFile.cpp
index 383bb89e8..706124e1b 100644
--- a/src/FileIO/FitRideFile.cpp
+++ b/src/FileIO/FitRideFile.cpp
@@ -462,6 +462,15 @@ struct FitFileReaderState
                 case 2: return "Pioneer SGX-CA500";
                 default: return QString("Pioneer %1").arg(prod);
             }
+        } else if (manu == 69) {
+            // Stages Cycling
+            switch (prod) {
+                case -1:  return "Stages Cycling";
+                case 1: return "Stages Power Gen 1";
+                case 2: return "Stages Power Gen 2";
+                case 3: return "Stages Power Gen 3";
+                default: return QString("Stages Cycling %1").arg(prod);
+            }
         } else if (manu == 70) {
             // does not set product at this point
            return "Sigmasport ROX";
-- 
2.23.0

