From 6979d665d73b91b3fcb1fcd4762dd1caf3ee82bd Mon Sep 17 00:00:00 2001
From: Luuk Hendriks <mail@luukhendriks.eu>
Date: Wed, 22 Aug 2018 00:54:43 +0200
Subject: [PATCH] missing ifdef GC_HAVE_ICAL

---
 src/Gui/LTMSidebar.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Gui/LTMSidebar.cpp b/src/Gui/LTMSidebar.cpp
index aceab25e8f..53a942ca0f 100644
--- a/src/Gui/LTMSidebar.cpp
+++ b/src/Gui/LTMSidebar.cpp
@@ -1279,10 +1279,12 @@ LTMSidebar::addEvent()
 
         active = true;
 
+#ifdef GC_HAVE_ICAL
         // upload to remote calendar if configured
         if (context->athlete->davCalendar->getConfig())
             if (!context->athlete->davCalendar->upload(&myevent))
                 QMessageBox::warning(this, tr("Add Event"), tr("The new event could not be uploaded to your remote calendar."));
+#endif
 
         seasons->seasons[seasonindex].events.append(myevent);
 
