--- Makefile	2012-06-10 09:44:49.439826346 +0200
+++ Makefile	2012-06-10 10:17:33.233564277 +0200
@@ -2,6 +2,7 @@
 
 KEYRINGFLAGS = `pkg-config --libs --cflags gnome-keyring-1`
 PURPLEFLAGS = `pkg-config --cflags purple`
+PLUGINDIR = `pkg-config --variable=plugindir purple`
 
 
 all: ${TARGET}.so
@@ -11,8 +12,8 @@
 	rm -rf pidgin-${TARGET}-*
 
 ${TARGET}.so: ${TARGET}.c
-	${CC} -Wall -I. -g -O2 ${TARGET}.c -o ${TARGET}.so -shared -fPIC -DPIC -ggdb ${PURPLEFLAGS} ${KEYRINGFLAGS}
+	${CC} ${CFLAGS} -Wall -I. ${TARGET}.c -o ${TARGET}.so -shared -fPIC -DPIC ${PURPLEFLAGS} ${KEYRINGFLAGS}
 
 install: ${TARGET}.so
-	mkdir -p ~/.purple/plugins
-	cp ${TARGET}.so ~/.purple/plugins/
+	mkdir -p ${DESTDIR}${PLUGINDIR}
+	cp ${TARGET}.so ${DESTDIR}${PLUGINDIR}
