diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9760347..a40a3d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,7 +175,7 @@ include(CPack)
 
 if (UNIX AND NOT MINGW AND NOT APPLE)
     install(TARGETS ${BINARY} RUNTIME DESTINATION bin)
-    install(DIRECTORY externalized mods _unittests DESTINATION share/ja2)
+    install(DIRECTORY externalized mods DESTINATION share/ja2)
     install(FILES _build/distr-files-linux/ja2-stracciatella.desktop DESTINATION share/applications)
     install(
         FILES _build/icons/logo.svg
