From f9e9cbd046d56d29807c3964097a04c03aaf1272 Mon Sep 17 00:00:00 2001
From: Poncho <poncho@spahan.ch>
Date: Tue, 11 Apr 2017 12:51:46 +0200
Subject: [PATCH 2/2] Only depend on release packages

---
 rust/Cargo.toml | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/rust/Cargo.toml b/rust/Cargo.toml
index 957ec3c..89f1a31 100644
--- a/rust/Cargo.toml
+++ b/rust/Cargo.toml
@@ -3,11 +3,6 @@ name = "stracciatella"
 version = "0.1.0"
 authors = ["Stefan Lau <github@stefanlau.com>"]
 
-[profile.dev]
-rpath = true
-[profile.release]
-rpath = true
-
 [lib]
 crate-type = ["dylib"]
 name = "stracciatella"
@@ -18,12 +13,3 @@ getopts = "0.2"
 libc = "0.2.16"
 serde = "0.8"
 serde_json = "0.8"
-
-[dev-dependencies]
-regex = "0.1"
-tempdir = "0.3"
-
-[target.'cfg(windows)'.dependencies]
-winapi = "0.2"
-user32-sys = "0.2"
-shell32-sys = "0.1"
-- 
2.10.2

