From 477d50169a42431d7fb0f939eb851ef6a2d80c3a Mon Sep 17 00:00:00 2001
From: Hiltjo Posthuma <hiltjo@codemadness.org>
Date: Fri, 22 Feb 2019 11:13:44 +0100
Subject: [PATCH 4/8] bump version to 0.9.1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 72e6213..aca1353 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 include config.mk
 
 NAME = stagit
-VERSION = 0.9
+VERSION = 0.9.1
 
 SRC = \
 	stagit.c\
-- 
2.21.0

