diff --unified -uNrp audiothumbnailer-0.2.0/src/audiothumbnailer.cpp audiothumbnailer-0.2.0-new/src/audiothumbnailer.cpp
--- audiothumbnailer-0.2.0/src/audiothumbnailer.cpp	2008-11-18 15:02:03.000000000 +0100
+++ audiothumbnailer-0.2.0-new/src/audiothumbnailer.cpp	2013-08-18 10:24:35.909964276 +0200
@@ -30,7 +30,7 @@
 #include <mpegfile.h>
 #include <attachedpictureframe.h>
 
-#include <ImageMagick/Magick++.h>
+#include <ImageMagick-6/Magick++.h>
 
 using namespace std;
 using namespace Magick;
