From f1913fd9607c4c91e749d534d7da07e1311b180e Mon Sep 17 00:00:00 2001
From: lexmiller <lex.miller@gmail.com>
Date: Sun, 6 Jun 2021 13:55:47 +0900
Subject: [PATCH 3/4] IMDB 2021 format: Fix origtitle

---
 lib/IMDB_Movie.pm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/IMDB_Movie.pm b/lib/IMDB_Movie.pm
index 444becf..d9bea48 100644
--- a/lib/IMDB_Movie.pm
+++ b/lib/IMDB_Movie.pm
@@ -411,6 +411,7 @@ sub _otitle {
     $tag = _jump_class($parser, "originalTitle") or return undef;
     $otitle = get_text_html($parser);
     $otitle = unqote_and_strip_space($otitle);
+    $otitle =~ s/^Original title: //;
     return $otitle;
 }
 
-- 
2.31.1

