From 1ddedc7ecbf847a42e306172906bce06090688e8 Mon Sep 17 00:00:00 2001
From: lexmiller <lex.miller@gmail.com>
Date: Sun, 6 Jun 2021 14:56:10 +0900
Subject: [PATCH 4/4] IMDB 2021 format: Fix user rating

---
 lib/IMDB_Movie.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/IMDB_Movie.pm b/lib/IMDB_Movie.pm
index d9bea48..173228c 100644
--- a/lib/IMDB_Movie.pm
+++ b/lib/IMDB_Movie.pm
@@ -704,8 +704,8 @@ sub _user_rating {
     my $tag;
     my $rating;
 
-    # <span itemprop="ratingValue">8.4</span>
-    _jump_prop($parser, "itemprop", "ratingValue", "span") or return undef;
+    # <span class="AggregateRatingButton__RatingScore-sc-1il8omz-1 fhMjqK">8.2</span>
+    _jump_prop($parser, "class", "AggregateRatingButton__RatingScore-sc-1il8omz-1 fhMjqK", "span") or return undef;
     $rating = $parser->get_text() or return undef;
     if ($rating =~ /([\d.]+)/) {
         return $1;
-- 
2.31.1

