--- src/Makefile.am.old	2012-08-14 13:34:40.299853824 +0200
+++ src/Makefile.am	2012-08-14 13:35:08.735498330 +0200
@@ -1,4 +1,4 @@
-CXXFLAGS = -Wall -O3 -DSYS_CONF_DIR=\"$(sysconfdir)\"
+AM_CXXFLAGS = $(CXXFLAGS) -Wall -DSYS_CONF_DIR=\"$(sysconfdir)\"
 
 bin_PROGRAMS = logkeys llk llkk
 logkeys_SOURCES = logkeys.cc
@@ -10,4 +10,4 @@
 	chmod u+s    $(DESTDIR)$(bindir)/llk
 	chown root\: $(DESTDIR)$(bindir)/llkk
 	chmod u+s    $(DESTDIR)$(bindir)/llkk
-  
\ No newline at end of file
+  
