diff --git a/Makefile b/Makefile
index 5a61eda..7a9c4ea 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ endif
 
 GNOME_CFLAGS     := `pkg-config --cflags gnome-keyring-1`
 GNOME_LDFLAGS    := `pkg-config --libs gnome-keyring-1`
-CXXFLAGS         += -Wall -fno-rtti -fno-exceptions -fPIC -std=gnu++0x -D__STDC_LIMIT_MACROS
+CXXFLAGS         += -Wall -fno-rtti -fno-exceptions -fPIC -std=gnu++0x -D__STDC_LIMIT_MACROS -Wno-invalid-offsetof -Wno-delete-non-virtual-dtor -Wno-maybe-uninitialized
 LDFLAGS          +=
 
 # if auto-detection fails, fall back to these values
