diff --unified --new-file --recursive --show-c-function timeline-1.10.0/source/timelinelib/config/paths.py timeline-1.10.0-fix/source/timelinelib/config/paths.py
--- timeline-1.10.0/source/timelinelib/config/paths.py	2016-04-30 06:53:27.000000000 +0200
+++ timeline-1.10.0-fix/source/timelinelib/config/paths.py	2016-05-02 10:01:26.795580978 +0200
@@ -19,7 +19,7 @@
 import os.path
 
 
-_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))
+_ROOT = "/usr/share/timeline/"
 _ROOT = _ROOT.decode("utf-8")
 ICONS_DIR = os.path.join(_ROOT, u"icons")
 EVENT_ICONS_DIR = os.path.join(_ROOT, u"icons", u"event_icons")
