--- HandBrake-0.9.6/test/module.defs	2011-10-21 20:24:43.000000000 +0200
+++ HandBrake-0.9.6-unbundle/test/module.defs	2012-06-02 14:09:04.943789898 +0200
@@ -10,10 +10,12 @@ TEST.c.o = $(patsubst $(SRC/)%.c,$(BUILD
 TEST.exe = $(BUILD/)$(call TARGET.exe,$(HB.name)CLI)
 
 TEST.libs = $(LIBHB.a) $(foreach n, \
-        a52 ass avcodec avformat avutil dca dvdnav dvdread faac fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
-        ogg samplerate swscale theora vorbis vorbisenc x264 xml2 bluray, \
+        a52 avcodec avformat avutil dca dvdnav dvdread faac mkv mpeg2 mp3lame mp4v2 \
+        ogg samplerate swscale theora vorbis vorbisenc x264 bluray, \
         $(CONTRIB.build/)lib/lib$(n).a )
 
+TEST.libs += -lass -lfontconfig -lfreetype -lxml2
+
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 
 ifeq (1,$(PTHREADW32.enabled))
