diff --git a/gtk/src/Makefile.am b/gtk/src/Makefile.am
index 9c1dbd2..d0115e9 100644
--- a/gtk/src/Makefile.am
+++ b/gtk/src/Makefile.am
@@ -10,7 +10,7 @@ else
 HB_LIBS= \
 	-lhb -la52 -lmkv -lavformat -lavcodec -lavutil -ldca -ldvdnav -ldvdread \
 	-lfaac -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
-	-lx264 -lmp4v2 -lswscale -ltheora -lz \
+	-lx264 -lmp4v2 -lswscale -ltheoraenc -ltheoradec -lz -lxml2 \
 	-lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype
 endif
 
diff --git a/make/include/main.defs b/make/include/main.defs
index e68f679..b8cb683 100644
--- a/make/include/main.defs
+++ b/make/include/main.defs
@@ -21,22 +21,11 @@ ifneq (,$(filter $(BUILD.system),darwin cygwin mingw))
 endif
 
 
-MODULES += contrib/faac
 MODULES += contrib/ffmpeg
-MODULES += contrib/fontconfig
-MODULES += contrib/freetype
-MODULES += contrib/lame
-MODULES += contrib/libass
 MODULES += contrib/libdca
 MODULES += contrib/libdvdread
 MODULES += contrib/libdvdnav
-MODULES += contrib/libbluray
 MODULES += contrib/libmkv
-MODULES += contrib/libogg
-MODULES += contrib/libsamplerate
-MODULES += contrib/libtheora
-MODULES += contrib/libvorbis
-MODULES += contrib/libxml2
 MODULES += contrib/mp4v2
 MODULES += contrib/mpeg2dec
 
@@ -46,8 +35,6 @@ ifneq ($(HAS.pthread),1)
 endif
 endif
 
-MODULES += contrib/x264
-
 ifneq (,$(filter $(BUILD.system),cygwin mingw))
 ifneq ($(HAS.iconv),1)
     MODULES += contrib/libiconv
diff --git a/test/module.defs b/test/module.defs
index 5030e6f..f05b71a 100644
--- a/test/module.defs
+++ b/test/module.defs
@@ -10,10 +10,12 @@ TEST.c.o = $(patsubst $(SRC/)%.c,$(BUILD/)%.o,$(TEST.c))
 TEST.exe = $(BUILD/)$(call TARGET.exe,$(HB.name)CLI)
 
 TEST.libs = $(LIBHB.a) $(foreach n, \
-        a52 ass avcodec avformat avutil dca dvdnav dvdread faac fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
-        ogg samplerate swscale theora vorbis vorbisenc x264 xml2 bluray, \
+        a52 avcodec avformat avutil dca dvdnav dvdread mkv mpeg2 mp4v2 swscale, \
         $(CONTRIB.build/)lib/lib$(n).a )
 
+TEST.libs += -lass -lfontconfig -lfreetype -lxml2 -lbluray -lsamplerate -logg -lmp3lame \
+        -lx264 -lvorbis -lvorbisenc -ltheoraenc -ltheoradec -lfaac
+
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 
 ifeq (1,$(PTHREADW32.enabled))
