diff --unified -uNrp mozilla-release/configure mozilla-release-fix/configure
--- mozilla-release/configure	2013-10-26 00:28:28.000000000 +0200
+++ mozilla-release-fix/configure	2013-11-03 11:32:16.461963440 +0100
@@ -1181,7 +1181,7 @@ _SUBDIR_HOST_LDFLAGS="$HOST_LDFLAGS"
 _SUBDIR_CONFIG_ARGS="$ac_configure_args"
 
 MOZJPEG=62
-MOZPNG=10517
+MOZPNG=10515
 NSPR_VERSION=4
 NSS_VERSION=3
 
diff --unified -uNrp mozilla-release/configure.in mozilla-release-fix/configure.in
--- mozilla-release/configure.in	2013-10-26 00:27:10.000000000 +0200
+++ mozilla-release-fix/configure.in	2013-11-03 11:32:45.449026855 +0100
@@ -51,7 +51,7 @@ _SUBDIR_CONFIG_ARGS="$ac_configure_args"
 dnl Set the version number of the libs included with mozilla
 dnl ========================================================
 MOZJPEG=62
-MOZPNG=10517
+MOZPNG=10515
 NSPR_VERSION=4
 NSS_VERSION=3
 
