diff --unified --new-file --recursive --show-c-function darksky_weather-1.6.5-fix/setup.py darksky_weather-1.6.5/setup.py
--- darksky_weather-1.6.5-fix/setup.py	2020-01-20 10:57:28.924729883 +0100
+++ darksky_weather-1.6.5/setup.py	2020-01-20 10:58:12.547668624 +0100
@@ -12,7 +12,7 @@ with open(os.path.join(os.path.abspath(o
 setup(
     version=__version__,
     name="darksky_weather",
-    packages=find_packages(),
+    packages=find_packages(exclude=['tests']),
     install_requires=["requests==2.21.0", "pytz==2019.1", "aiohttp==3.5.4"],
     description="The Dark Sky API wrapper",
     long_description="View on github",
